cdap-kubernetes/src/main/java/co/cask/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -0,0 +1,474 @@
+  }
+
+  @Override
+  public TwillPreparer withResources(Iterable<URI> resources) {
+    for (URI resource : resources) {
+      this.resources.add(resource);
+    }
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withClassPaths(String... classPaths) {
+    return withClassPaths(Arrays.asList(classPaths));
+  }
+
+  @Override
+  public TwillPreparer withClassPaths(Iterable<String> classPaths) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withEnv(Map<String, String> env) {
+    for (String runnableName : runnables) {
+      withEnv(runnableName, env);
+    }
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withEnv(String runnableName, Map<String, String> env) {
+    environments.put(runnableName, env);
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withApplicationClassPaths(String... classPaths) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withApplicationClassPaths(Iterable<String> classPaths) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withBundlerClassAcceptor(ClassAcceptor classAcceptor) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withMaxRetries(String runnableName, int maxRetries) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer addSecureStore(SecureStore secureStore) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer setLogLevel(LogEntry.Level logLevel) {
+    return setLogLevels(Collections.singletonMap(Logger.ROOT_LOGGER_NAME, logLevel));
+  }
+
+  @Override
+  public TwillPreparer setLogLevels(Map<String, LogEntry.Level> logLevels) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer setLogLevels(String runnableName, Map<String, LogEntry.Level> logLevelsForRunnable) {
+    return this;
+  }
+
+  @Override
+  public TwillPreparer setClassLoader(String classLoaderClassName) {
+    return this;
+  }
+
+  @Override
+  public TwillController start() {
+    return start(60, TimeUnit.SECONDS);
+  }
+
+  @Override
+  public TwillController start(long timeout, TimeUnit timeoutUnit) {
+    try {
+      V1ConfigMap configMap = createConfigMap();
+      createDeployment(configMap);
+      return new KubeTwillController(resourceMeta.getName(), kubeNamespace, runId, apiClient, discoveryServiceClient);
+    } catch (ApiException | IOException e) {
+      throw new RuntimeException("Unable to create Kubernetes resource while attempting to start program.", e);
+    }
+  }
+
+  private V1Deployment createDeployment(V1ConfigMap configMap) throws ApiException {
+    /*
+       The deployment will look something like:
+       apiVersion: apps/v1
+       kind: Deployment
+       metadata:
+         [deployment meta]
+       spec:
+         [deployment spec]
+       template:
+         [template spec]
+     */
+    AppsV1Api appsApi = new AppsV1Api(apiClient);
+    V1Deployment deployment = new V1Deployment();
+
+    /*
+       metadata:
+         name: service-runid
```

Comment:
Better prefix the deployment name with `cdap-<instanceName>`, where the `instanceName` is already grabbed by the `KubeMasterEnvironment`.